spec/ameba/rule/lint/unused_comparison_spec.cr

```diff
@@ -0,0 +1,172 @@
+          x === 2
+        # ^^^^^^^ error: Comparison operation is unused
+          x < 2
+        # ^^^^^ error: Comparison operation is unused
+          x <= 2
+        # ^^^^^^ error: Comparison operation is unused
+          x > 2
+        # ^^^^^ error: Comparison operation is unused
+          x >= 2
+        # ^^^^^^ error: Comparison operation is unused
+          x <=> 2
+        # ^^^^^^^ error: Comparison operation is unused
+        puts x
+        CRYSTAL
+    end
+
+    it "fails for an unused top-level comparison" do
+      expect_issue subject, <<-CRYSTAL
+        x = 1
+          x == 2
+        # ^^^^^^ error: Comparison operation is unused
+        puts x
+        CRYSTAL
+    end
+
+    it "fails for an unused comparison in a begin block" do
+      expect_issue subject, <<-CRYSTAL
+        begin
+          x = 1
+          x == 2
+        # ^^^^^^ error: Comparison operation is unused
+          puts x
+        end
+        CRYSTAL
+    end
+
+    it "fails for unused comparisons in if/elsif/else bodies" do
+      expect_issue subject, <<-CRYSTAL
+        a = if x = 1
+              x == 1
+            # ^^^^^^ error: Comparison operation is unused
+              x == 2
+            elsif true
+              x == 1
+            # ^^^^^^ error: Comparison operation is unused
+              x == 2
+            else
+              x == 2
+            # ^^^^^^ error: Comparison operation is unused
+              x == 1
+            # ^^^^^^ error: Comparison operation is unused
+              x == 3
+            end
+        CRYSTAL
+    end
+
+    it "fails for unused comparisons in a proc body" do
+      expect_issue subject, <<-CRYSTAL
+        a = -> {
+          x == 1
+        # ^^^^^^ error: Comparison operation is unused
+          "meow"
+        }
+        CRYSTAL
+    end
+
+    it "fails for unused comparison in if when not assigning" do
+      expect_issue subject, <<-CRYSTAL
+        if true
+          x == 1
+        # ^^^^^^ error: Comparison operation is unused
+        else
+          x == 2
+        # ^^^^^^ error: Comparison operation is unused
+        end
+        CRYSTAL
+    end
+
+    it "fails on useless comparisons" do
+      expect_issue subject, <<-CRYSTAL
+        def hello
+          if x == 3
+            x < 1
+          else
+            x > 1
+          end
+        end
+
+        def world
+          if x == 3
+            x < 1
+          # ^^^^^ error: Comparison operation is unused
+          else
+            x > 1
+          # ^^^^^ error: Comparison operation is unused
+          end
+
+          return
+        end
+
+        if x == 3
+          x < 1
+        # ^^^^^ error: Comparison operation is unused
+        else
+          x > 1
+        # ^^^^^ error: Comparison operation is unused
+        end
+
+        a = if x == 3
+              x > 1
+            # ^^^^^ error: Comparison operation is unused
+              x < 1
+            else
+              x > 1
+            end
```

Comment:
why the other two are not detected here?

Reply:
Because they're assigned to the value